src/server/TCPServer.cpp

```diff
@@ -191,17 +218,63 @@ void TCPServer::HandlePrepareAck(const message::PrepareAck &m,
 
 void TCPServer::HandlePropose(const message::Propose &m,
     TCPConnection::pointer connection) {
-  round_->HandlePropose(m, connection);
+  auto round = GetRound(m.GetBallotNum());
```

Comment:
Consider making a generic HandleMessage helper that these all call.

Reply:
Not quite sure what you mean here. Clarify?

Reply:
Ok I think I see what you mean -- it isn't possible under the current architecture. Messages do not inherit from any common superclass, so there has to be a separate handler per message type for now.